scipy/stats/tests/test_morestats.py

```diff
@@ -1669,36 +1673,38 @@ def test_permutation_method(self, size):
            12.10, 15.02, 16.83, 16.98, 19.92, 9.47, 11.68, 13.41, 15.35, 19.11]
 
 
+@array_api_compatible
 class TestKstat:
-    def test_moments_normal_distribution(self):
+    def test_moments_normal_distribution(self, xp):
         np.random.seed(32149)
-        data = np.random.randn(12345)
-        moments = [stats.kstat(data, n) for n in [1, 2, 3, 4]]
+        data = xp.asarray(np.random.randn(12345), dtype=xp.float64)
```

Comment:
Even `torch` will produce a `float64` Tensor if it is generated from a NumPy `float64` array, right?

Reply:
Yes, I can't quite remember why I did this, will have a look in the follow up